Tests/NIOTests/DatagramChannelTests.swift

```diff
@@ -407,4 +407,37 @@ final class DatagramChannelTests: XCTestCase {
     public func testSetGetOptionClosedDatagramChannel() throws {
         try assertSetGetOptionOnOpenAndClosed(channel: firstChannel, option: ChannelOptions.maxMessagesPerRead, value: 1)
     }
+
+    func testOpenWithoutBind() throws {
+        class ActiveVerificationHandler : ChannelInboundHandler {
+            typealias InboundIn = AddressedEnvelope<ByteBuffer>
+
+            private var activeReceived = false
+            public func channelActive(ctx: ChannelHandlerContext) {
+                self.activeReceived = true
+            }
+
+            public func channelInactive(ctx: ChannelHandlerContext) {
+                XCTAssertTrue(self.activeReceived)
+            }
+        }
```

Comment:
This test would pass if `channelActive` never fires, as this assert would never run.